go/vt/vtgate/executor.go

```diff
@@ -683,6 +683,14 @@ func (e *Executor) handleShow(ctx context.Context, safeSession *SafeSession, sql
 	defer func() { logStats.ExecuteTime = time.Since(execStart) }()
 
 	switch show.Type {
+	case sqlparser.KeywordString(sqlparser.VARIABLES):
+		if destKeyspace == "" {
+			keyspaces, err := e.resolver.resolver.GetAllKeyspaces(ctx)
+			if err !=nil {
```

Comment:
Need to check that at least one keyspace was returned.

Reply:
Right! Making the change now.